image-registry/remove-registry-baremetal.md

```diff
@@ -0,0 +1,205 @@
+loss if a permanent form of storage is not configured.
+
+### Goals
+
+- `openshift-installer` completes if the platform is set to `baremetal` or
+  `oVirt`.
+- UPI installs for `vSphere` bring the registry up in the `Removed` state.
+- Image registry is able to bring itself up after the management state is
+  switched from `Removed` to `Managed`.
+
+### Non-Goals
+
+- Provision storage for the registry for RHV and k8s-native infrastructre.
+- Provide 1st class support for additional storage types/providers for the
+  registry (ex: MCG).
+- Harden OpenShift features (ex: Builds, ImageStreams) such that they error
+  cleanly if the registry is not installed.
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal actually is.
+
+### User Story
+
+As an OpenShift cluster admin/installer
+I want the registry to be `Removed` on RHV and k8s-native infra installations
+So that I can provision other storage providers
+And later enable the image registry with the desired storage
+
+### Implementation Details/Notes/Constraints
+
+- When the image registry operator bootstraps itself, it will inspect the
+  cluster infrastructure platform status.
+- The registry will bootstrap itself as `Removed` on the following platform types:
+  - `BareMetal` (i.e. baremetal platforms that support IPI install)
+  - `oVirt` (RHV)
+  - `VSphere`
+  - `None` (true UPI baremetal installs)
+- The registry will bootstrap itself as `Managed` with `EmptyDir` storage on
+  the following platform types:
+  - `Libvirt`
+  - `Unknown` platforms
+- The registry will continue to bootstrap itself as `Managed` with specific
+  storage on the following platforms:
+  - `AWS` (Amazon S3)
+  - `Azure` (Azure blob storage)
+  - `GCP` (Google Cloud Storage)
+  - `OpenStack` (Swift)
+- Any platform that wishes to set up the registry must perform a day 2 action
+  that does the following:
+  - Provisions suitable storage, typically a PV that supports the `RWX` access
+    mode.
+  - Updates the image registry operator's management state to `Managed`.
+- Telemeter should be aware if a cluster's registry has been removed:
+  - Reason codes in the ClusterOperator conditions (`Available` in particular).
+  - Prometheus alerts warning the registry has been removed.
+
+### Risks and Mitigations
+
+#### Image registry is never installed
+
+Several components will likely report errors if the registry is `Removed` and
+an admin does nothing to set up the registry. Potential candidates that will
+break:
+
+1. Builds which push to an imagestream tag will error at runtime.
+2. Samples operator imagestreams which import from `registry.redhat.io` will
+   require a pull secret to be usable.
+3. Imagestreams imported with `--reference-policy=local` will not reference the
+   internal registry as expected. `BuildConfig` and `DeploymentConfig` objects
+   which do their own resolution of `ImageStreamTag` will break.
```

Comment:
i don't know that they'll break, but they'll resolve to the external pullspec(instead of the internal registry pullspec) which may require creds that the build/deployment doesn't have available.

Reply:
@bparees updated to clarify failure mode (`ImagePullBackoff`)